api/api_test.go

```diff
@@ -18,11 +19,24 @@ func TestSetup(t *testing.T) {
 		ctx := context.Background()
 
 		mockMongoDB := &mock.DataStoreMock{}
-		cacheAPI := api.Setup(ctx, router, mockMongoDB)
 
-		Convey("When created the following routes should have been added", func() {
-			So(hasRoute(cacheAPI.Router, "/v1/cache-times/{id}", "GET"), ShouldBeTrue)
-			So(hasRoute(cacheAPI.Router, "/v1/cache-times/{id}", "PUT"), ShouldBeTrue)
+		Convey("When created in publishing mode", func() {
+			cfg := &config.Config{IsPublishing: true}
```

Comment:
You don't need to create a full Config struct for this. A simple boolean variable would do: `isPublishing := true`. The same goes for the test below.